mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala

```diff
@@ -188,4 +189,76 @@ object ImageSchema {
       Some(Row(Row(origin, height, width, nChannels, mode, decoded)))
     }
   }
+
+  /**
+   * Read the directory of images from the local or remote source
+   *
+   * @note If multiple jobs are run in parallel with different sampleRatio or recursive flag,
+   * there may be a race condition where one job overwrites the hadoop configs of another.
+   * @note If sample ratio is less than 1, sampling uses a PathFilter that is efficient but
+   * potentially non-deterministic.
+   *
+   * @param path Path to the image directory
+   * @return DataFrame with a single column "image" of images;
+   *         see ImageSchema for the details
+   */
+  @deprecated("use `spark.read.format(\"image\").load(path)` and this `readImages` will be " +
+    "removed in 3.1.0.", "2.4.0")
```

Comment:
In any way, this is wrong due to the misunderstanding. 

With the same reason why you add this back, @marmbrus 's suggestion was not removing this forever.